spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -258,6 +258,9 @@ def main(argv=None):
         # Reparse the arguments
         arguments = parser.parse_args(argv)
 
+    if arguments.script is None:
+        parser.error("The -script argument must be provided, either via command-line or via the -config/-c argument.")
```

Comment:
This is simpler than trying to trick argparse into calling `error()` for us, which is the ultimate goal here: I initially added `required=True` to `-i` to fix https://github.com/neuropoly/spinalcordtoolbox/issues/3154, but that didn't work because these scripts have pseudo-subcommands, and that would cause it to `error()` before we could check for them.

If we were already following the GNU conventions then I would say we should rewrite to use `argparse.add_subparsers()`. But we're not. So yeah, let's do this!